docs/gradle-plugin.md

```diff
@@ -0,0 +1,270 @@
+
+    Make sure the jte gradle plugin version matches the jte dependency version. You can create a `jteVersion` in `gradle.properties` to sync the versions easily.
+
+The plugin provides two tasks:
+
+## `precompileJte` task { #precompile-task }
+
+See [Precompiling Templates](pre-compiling.md) for more information. To use and configure this task, you would use:
+
+=== "Groovy"
+
+    ```groovy linenums="1"
+    plugins {
+        id 'java'
+        id 'gg.jte.gradle' version '${jteVersion}'
+    }
+
+    dependencies {
+        implementation('gg.jte:jte:${jteVersion}')
+    }
+
+    jte {
+        precompile()
+    }
+    ```
+
+=== "Kotlin"
+
+    ```kotlin linenums="1"
+    plugins {
+        java
+        id("gg.jte.gradle") version("${jteVersion}")
+    }
+
+    dependencies {
+        implementation("gg.jte:jte:${jteVersion}")
+    }
+
+    jte {
+        precompile()
+    }
+    ```
+
+### Task inputs { #precompile-inputs }
+
+The follow inputs will be picked by the `precompileJte` task:
+
+| Parameter               | Description                                                                       | Default                                                        |
+|-------------------------|-----------------------------------------------------------------------------------|----------------------------------------------------------------|
+| `sourceDirectory`       | The directory where template files are located                                    | `src/main/jte`                                                 |
+| `targetDirectory`       | The directory where compiled classes should be written to                         | `jte-classes`                                                  |
+| `compilePath`           | The compile-classpath to use                                                      | `sourceSets.main.runtimeClasspath`                             |
+| `contentType`           | The content type of all templates. Either `Plain` or `Html`                       | `Html`                                                         |
+| `trimControlStructures` | Trims control structures, resulting in prettier output                            | `false`                                                        |
+| `htmlTags`              | Intercepts the given html tags during template compilation                        | None                                                           |
+| `htmlPolicyClass`       | An [`HtmlPolicy`][html-policy] used to check the parsed HTML at compile time      | None                                                           |
+| `htmlCommentsPreserved` | If HTML comments should be preserved in the output                                | `false`                                                        |
+| `binaryStaticContent`   | If to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
```

Comment:
```suggestion
| `htmlCommentsPreserved` | Whether HTML comments should be preserved in the output                                | `false`                                                        |
| `binaryStaticContent`   | Whether to generate a [binary content](binary-rendering.md) resource for each template | `false`                                                        |
```

Reply:
Not sure about these, but `if to` sounds really strange.

Reply:
English is not my first language, so it may indeed be strange. However, it is consistent with the javadoc comments. If we decide to adjust here, I would rather adjust the javadoc comments and the Maven plugin docs page.

Reply:
I'm not a native speaker either.

Reply:
Grammarly seems okay with this phrasing. I think we can leave it like this.